js/src/controllers/media-controller.js

```diff
@@ -47,6 +47,49 @@ var MediaController = wp.media.controller.State.extend({
 		this.props.set( 'search', null );
 	},
 
+	setActionSelect: function() {
+		this.attributes.title = shortcodeUIData.strings.media_frame_menu_insert_label;
+		this.props.set( 'currentShortcode', null );
+		this.props.set( 'action', 'select' );
+
+		// Update menuItem text.
+		var menuItem = this.frame.menu.get().get('shortcode-ui');
+		menuItem.options.text = this.attributes.title;
+		menuItem.render();
+
+		this.frame.setState( 'shortcode-ui' );
+		this.frame.render();
+	},
+
+	setActionUpdate: function( currentShortcode ) {
+
+		this.attributes.title = shortcodeUIData.strings.media_frame_menu_update_label;
+		this.attributes.title = this.attributes.title.replace( /%s/, currentShortcode.attributes.label );
+
+		this.props.set( 'currentShortcode', currentShortcode );
+		this.props.set( 'action', 'update' );
+
+		// If a new frame is being created, it may not exist yet.
+		// Defer to ensure it does.
+		_.defer( function() {
+
+			this.frame.setState( 'shortcode-ui' );
+			this.frame.content.render();
+
+			this.toggleSidebar( true );
+
+			this.frame.once( 'close', function() {
+				this.frame.mediaController.toggleSidebar( false );
+			}.bind( this ) );
+
+		}.bind( this ) );
+
+	},
+
+	toggleSidebar: function( show ) {
```

Comment:
Very minor nit - could this variable be named `hide` rather than `show`, since it determines whether or not the sidebar will be hidden.